homeassistant/components/touchline/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Config flow for Roth Touchline floor heating controller."""
+from __future__ import annotations
+
+import logging
+import re
+from typing import Any
+
+from pytouchline import PyTouchline
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+RESULT_SUCCESS = "success"
+RESULT_CANNOT_CONNECT = "cannot_connect"
+
+
+def _try_connect_and_fetch_basic_info(host):
+    """Attempt to connect and, if successful, fetch number of devices."""
+    py_touchline = PyTouchline()
+    result = {"type": None, "data": {}}
+    number_of_devices = None
+    try:
+        number_of_devices = int(py_touchline.get_number_of_devices(host))
+        if number_of_devices:
+            result["type"] = RESULT_SUCCESS
```

Comment:
Wouldn't it make logical sense to set the result "type" only when we're sure we're successful? With this, it always gets set to `RESULT_SUCCESS` first, even if the subsequent logic fails.

Reply:
To that end, we shouldn't put all of this inside the try/except—limit what goes inside to the actual code that can raise an exception.

Reply:
I have moved RESULT_SUCCES so that is only set, if w get a controller ID. Can I move more out of the try/except?